src/Sentry/Internal/DebugStackTrace.cs

```diff
@@ -346,18 +346,28 @@ internal static SentryStackFrame ParseNativeAOTToString(string info)
 
         if (stackFrame.GetFileName() is { } frameFileName)
         {
-            if (AttributeReader.TryGetProjectDirectory(method.Module.Assembly) is { } projectPath
-                && frameFileName.StartsWith(projectPath, StringComparison.OrdinalIgnoreCase))
+            var assembly = method.Module.Assembly;
+            var root = TryGetRoot(assembly, frameFileName, AttributeReader.TryGetRepositoryRoot)
+                ?? TryGetRoot(assembly, frameFileName, AttributeReader.TryGetProjectDirectory);
```

Comment:
@vaind I'm not sure if this is a good idea or not. My thinking was we'd try to resolve a git repository relative path and, if we couldn't do that, we'd try a project relative path (the later being "more specific").

A repo relative path would be useful for source link and code locations (in the delightful developer metrics stuff). 

What was the motivation for the original code that you wrote though, using a project relative path for the `frame.FileName`? Is what I've done here going to break anything there?